torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -23,7 +26,12 @@ DistAutogradContainer& DistAutogradContainer::init(int64_t worker_id) {
   auto& container = getInstance();
   container.worker_id_ = worker_id;
   container.next_context_id_ = static_cast<int64_t>(worker_id)
-      << kContextIdBits;
+      << kAutoIncrementBits;
+  container.next_autograd_message_id_ = static_cast<int64_t>(worker_id)
+      << kAutoIncrementBits;
+  container.max_id_ =
+      (kAutoIncrementMask |
+       (static_cast<int64_t>(worker_id) << kAutoIncrementBits));
   container.initialized_ = true;
```

Comment:
init() isn't thread safe nor is it safeguarding from init called more than once. I am assuming this is intentional, is it ?